procrastinate/worker.py

```diff
@@ -233,7 +233,7 @@ async def ensure_async() -> Callable[..., Awaitable]:
                 if inspect.iscoroutinefunction(task.func):
                     await_func = task
                 else:
-                    await_func = functools.partial(utils.sync_to_async, task)
+                    await_func = functools.partial(asyncio.to_thread, task)
```

Comment:
> It will be interesting to discover if doing that causes issues with Django applications (https://docs.djangoproject.com/en/5.1/topics/async/#async-safety).

I don't think so, as the worker manages its own connections (independent of Django), and only the user code would use Django connections.

Reply:
I was referring to the user code (the task implementation) that is being wrapped here.

What happens the when the user code uses Django connections?
How can they make it work when the thread the code runs in is not the Django main thread?

Reply:
From my understanding, it shouldn't matter if the Django connection wasn't initially created in the main thread but directly in the user thread. There is also no request/response cycle like in a Django view that messes around with those connections. And as you already recognized, our previous `sync_to_async` (with back then `thread_sensitive=True`) also didn't use the main thread [as the Django command never started the worker with `async_to_sync`](https://github.com/procrastinate-org/procrastinate/blob/b82b509c9dd03c50017505a7dec6969e2d2971c7/procrastinate/contrib/django/management/commands/procrastinate.py#L35). All sync tasks (back then when `thread_sensitive` was `True`) used a single thread, but it wasn't the main thread.
Since we switched to `thread_sensitive=False,` all sync tasks use their own separate thread. This should also work as [Django connections are thread local](https://medium.com/nerd-for-tech/djangos-database-connection-management-ad39a298d2e3). It should work as long as the user doesn't start threads on their own and reuses the connections there (the same as in Django itself).

Reply:
Ah yes, as long as we don't try to support `thread_sensitive=true`, that should be fine.

Reply:
Yes, and I also think that using `asyncio.to_thread` is, in our scenario, the same as using `sync_to_async` with `thread_sensitive=False` as we always started our worker in the Django command with `asyncio.run` and never used `async_to_sync` there.